src/components/Navbar.tsx

```diff
@@ -0,0 +1,183 @@
+    <li
+      tabIndex={0}
+      className="navbar-item"
+      onClick={handleClick}
+      onKeyDown={handleKeyDown}
+      ref={itemRef}
+    >
+      <FontAwesomeIcon
+        className="navbar-item-icon"
+        icon={icon}
+        size="2x" // this should be a param
+        color={
+          isSelected
+            ? style.primary
+            : theme?.colorMode === "dark"
+            ? style.lightBackground
+            : style.darkBackground
+        }
+      />
+      <div className={`navbar-item-title ${isSelected ? "selected" : ""}`}>
+        <span className="navbar-item-title-text">{title}</span>
+      </div>
+    </li>
+  );
+}
+
+type NavbarProps = {
+  label: string;
+  children: React.ReactNode;
+};
+
+export function Navbar(props: NavbarProps) {
+  const [selectedId, setSelectedId] = useState<number>();
+  const [items, setItems] = useState<React.RefObject<HTMLLIElement>[]>([]);
+  const selectedIndicatorRef = useRef<HTMLDivElement>(null);
+
+  // TODO: refactor this monstrosity (maybe ask Sandrina to give some tips)
+  const animateWithCss = (
+    domMenuItem: HTMLLIElement,
+    domSelectedIndicator: HTMLDivElement
+  ) => {
+    // First time an item is selected, the icon appears
+    if (!domSelectedIndicator.style.visibility) {
+      domSelectedIndicator.style.top = `${domMenuItem.offsetHeight - 12}px`;
+      domSelectedIndicator.style.left = `${
+        // center positioning the icon
+        domMenuItem.offsetLeft +
+        domMenuItem.clientWidth / 2 -
+        domSelectedIndicator.clientWidth / 2
+      }px`;
+      domSelectedIndicator.style.visibility = "visible";
+    } else {
+      // If an item is already selected, move the icon to next position
+      const currentPosition = domSelectedIndicator.offsetLeft;
+      const nextLeft = domMenuItem.offsetLeft;
+      const movement = nextLeft - currentPosition;
+      domSelectedIndicator.style.setProperty(
+        "--translate-x-value",
+        `${
+          movement +
+          domMenuItem.clientWidth / 2 -
+          domSelectedIndicator.clientWidth / 2
+        }px`
+      );
+      domSelectedIndicator.classList.add("animate");
+    }
+  };
+
+  useLayoutEffect(() => {
+    const domMenuItem = items[selectedId!]?.current;
+    const domSelectedIndicator = selectedIndicatorRef.current;
+    if (domMenuItem && domSelectedIndicator) {
+      animateWithCss(domMenuItem, domSelectedIndicator);
+    }
+  }, [items, selectedId]);
+
+  const registerItem = useCallback((item: React.RefObject<HTMLLIElement>) => {
+    setItems((prevItems) => [...prevItems, item]);
+  }, []);
+
+  return (
+    <ThemeProvider>
+      <NavbarContext.Provider
+        value={{
+          selectedId,
+          setSelectedId,
+          registerItem,
+        }}
+      >
+        <nav aria-label={props.label}>
+          <ul role="menubar" className="navbar">
+            {React.Children.map(props.children, (child, index) => {
+              if (isNavbarItem(child)) {
```

Comment:
praise: Well done with this check! 

One extra tip: In real projects, it's common to use env vars to conditionally throw these errors. For example, if you notice, React has a bunch of dev warnings that don't exist in production. This way the app doesn't explode in prod, and the bundle size is smaller (thanks to three-shaking)

You can check [vite's env vars](https://vitejs.dev/guide/env-and-mode.html), and apply it like this:


```
// utils.js
const isDev = () => import.meta.env.MODE === "development"

if (isDev() && isNavbarItem(child)) { ... }
```

Reply:
added, also fixed the check that had a bug on a closer look 😵‍💫